### PR TITLE
Parallel DynamicQuantizeLinear, QuantizeLinear and DequantizeLinear

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.
I used the MicroBenchmark to tune the TensorOpCost on my local machine with 6 cores. For small size (less than 80000), it uses single thread. For size(80000~100000), I see ~2x speedup. For very large data size, I see up to ~5x speedup.
**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/dynamicquantizelinear.cc [hunk 1]
```diff
@@ -66,7 +69,11 @@ Status DynamicQuantizeLinear<T>::Compute(OpKernelContext* ctx) const {
 
   // quantize the data
   auto* output = y.template MutableData<T>();
-  MlasQuantizeLinear(x_data, output, num_of_elements, scale, zero_point);
+
+  concurrency::ThreadPool* tp = ctx->GetOperatorThreadPool();
```
Comment (zhanghuanrong) [inline onnxruntime/core/providers/cpu/tensor/dynamicquantizelinear.cc:73]: Why not also parallel above two:
  auto min = ConstEigenVectorMap<float>(x_data, num_of_elements).minCoeff();
  auto max = ConstEigenVectorMap<float>(x_data, num_of_elements).maxCoeff();

File: onnxruntime/core/providers/cpu/tensor/quantize_linear.cc [hunk 2]
```diff
@@ -32,43 +34,50 @@ Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
...
-  float* output = y.template MutableData<float>();
 
-  for (size_t n = 0; n < static_cast<size_t>(N); n++) {
-    for (size_t bd = 0; bd < static_cast<size_t>(broadcast_dim); bd++) {
-      auto zp = static_cast<int32_t>(zero_point[bd]);
-      auto sc = scale[bd];
-
-      for (size_t bs = 0; bs < static_cast<size_t>(block_size); bs++) {
-        *output++ = static_cast<float>(static_cast<int32_t>(*input++) - zp) * sc;
+    concurrency::ThreadPool::TryParallelFor(tp, x_shape.Size(), 2.0 /*cost*/, [&](std::ptrdiff_t begin, std::ptrdiff_t end) {
```
Comment (zhanghuanrong) [inline onnxruntime/core/providers/cpu/tensor/quantize_linear.cc:72]: Default is hard for people to read or figure out.

Comment (zhanghuanrong) [reply onnxruntime/core/providers/cpu/tensor/quantize_linear.cc:72]: default [&]

File: onnxruntime/core/providers/cpu/tensor/dynamicquantizelinear.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/core/providers/cpu/tensor/dynamicquantizelinear.cc:73]: will add SIMD for the min/max reduce in a separate PR.

File: onnxruntime/core/providers/cpu/tensor/quantize_linear.cc [hunk 3]
```diff
@@ -32,43 +34,53 @@ Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
...
 
     // if an axis was specified, ensure the scale and zero point are compatible
     ORT_ENFORCE(x_scale.Shape().NumDimensions() == 1 && x_scale.Shape().Size() == broadcast_dim, "x_scale must be 1D tensor with size ", broadcast_dim);
     ORT_ENFORCE(x_zero_point.Shape().NumDimensions() == 1 && x_zero_point.Shape().Size() == broadcast_dim, "x_zero_point must be 1D tensor with size ", broadcast_dim);
-  } else {
-    N = 1;
-    broadcast_dim = 1;
-    block_size = static_cast<size_t>(x_shape.Size());
 
+    concurrency::ThreadPool::TryBatchParallelFor(
```
Comment (snnn) [inline onnxruntime/core/providers/cpu/tensor/quantize_linear.cc:53]: Probably you should put a lower bound there. 
```C++
if(N * broadcast_dim < xxx )
   run_in_single_thread();
else
    TryBatchParallelFor(...)  
```

**Pull request closed without merging**